specification/specification.yml

```
@@ -0,0 +1,751 @@
+
+    delete:
+      tags:
+        - offers
+      summary: Удаление предложения по аренде
+      description: Удаление предложения по аренде
+
+      parameters:
+        - name: offerId
+          in: path
+          required: true
+          schema:
+            type: string
+
+      responses:
+        '204':
+          description: Предложение и комментарии к нему были удалены.
+
+        '401':
+          description: Ошибка авторизации пользователя.
+
+        '404':
+          description: Предложение с указанным id не найдено.
+
+    get:
+      tags:
+        - offers
+      summary: Детальная информация по предложению
+      description: Детальная информация по предложению
+
+      parameters:
+        - name: offerId
+          in: path
+          required: true
+          schema:
+            type: string
+
+      responses:
+        '200':
+          description: Предложение найдено. В теле ответа информация о предложении.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/offer'
+
+        '404':
+          description: Предложение с указанным id не найдено.
+
+  /offers/premium/{cityName}:
+    get:
+      tags:
+        - offers
+      summary: Премиальные предложения для города
+      description: Получение списка премиальных предложений для определённого города. Премиальное предложение — предложение с флагом «Premium».
+
+      parameters:
+        - name: cityName
+          in: path
+          required: true
+          schema:
+            $ref: '#/components/schemas/offerCity'
+
+      responses:
+        '200':
+          description: В теле ответа информация о найденных предложениях.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/offer'
+
+  /offers/favorite:
+    get:
+      tags:
+        - offers
+      summary: Список избранных предложений
+      description: Список предложений, добавленных в избранное.
+
+      responses:
+        '200':
+          description: В теле ответа информация о найденных предложениях.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/offer'
+
+        '401':
+          description: Ошибка авторизации пользователя.
+
+    patch:
+      tags:
+        - offers
+      summary: Добавление и удаление предложения в/из избранного
+      description: Добавление и удаление предложения в/из избранного
+
+      responses:
+        '201':
+          description: Предложение было добавлено в избранное.
+
+        '401':
+          description: Ошибка авторизации пользователя.
+
+        '404':
+          description: Предложение с указанным id не найдено.
+
+        '422':
+          description: Ошибка валидации данных
+
+  /comments/{offerId}:
```

В URL получения и добавления комментариев для предложения, `comments` должен быть последнем аргументам так как, это та сущность которую  мы получаем / мутируем.
Если нам нужен другой id, `offerId`, обьязательно нужно писать названия сущность, `offers`,  перед ним. Порядок важен

```suggestion
  /offers/{offerId}/comments:
```